src/components/state-history-chart-line.html

```diff
@@ -1,265 +1,269 @@
-        var domain = states.domain;
-        var name = states.name;
-        var data = [];
-        var dataTable = new window.google.visualization.DataTable();
-        // array containing [time, value1, value2, etc]
-        var prevValues;
-        var hasTargetRange;
-        var processState;
-        var noInterpolations;
-        var series;
-        dataTable.addColumn({ type: 'datetime', id: 'Time' });
+        addColumn(name + ' current temperature');
+        addColumn(name + ' heating', 'area');
 
-        function pushData(values, noInterpolationValues) {
-          var timestamp = values[0];
-          if (timestamp > endTime) {
-            // Drop datapoints that are after the requested endTime. This could happen if
-            // endTime is "now" and client time is not in sync with server time.
-            return;
-          }
-          if (prevValues && noInterpolationValues) {
-            // if we have to prevent interpolation, we add an old value for each
-            // value that should not be interpolated at the same time that our new
-            // line will be published.
-            data.push([timestamp].concat(prevValues.slice(1).map(function (val, index) {
-              return noInterpolationValues[index] ? val : null;
-            })));
-          }
-          data.push(values);
-          prevValues = values;
+        if (hasTargetRange) {
+          addColumn(name + ' target temperature high', 'step');
+          addColumn(name + ' target temperature low', 'step');
+        } else {
+          addColumn(name + ' target temperature', 'step');
         }
-
-        if (domain === 'thermostat' || domain === 'climate') {
-          // We differentiate between thermostats that have a target temperature
-          // range versus ones that have just a target temperature
-          hasTargetRange = states.states.some(state => state.attributes &&
-            state.attributes.target_temp_high !== state.attributes.target_temp_low);
-
-          dataTable.addColumn('number', name + ' current temperature');
-          dataTable.addColumn('number', name + ' heating');
-          // The "heating" series uses steppedArea to shade the area below the current
-          // temperature when the thermostat is calling for heat.
-          // Its series index is 2 less than its column number--1 because
-          // zero-based and 1 because the first becomes the horizontal axis.
-          var seriesIndex = dataTable.getNumberOfColumns() - 2;
-          // Get existing series config, if there is any, rather than clobbering
-          options.series = Object.assign({}, options.series);
-          options.series[seriesIndex] = { type: 'steppedArea' };
-
+        processState = function (state) {
+          if (!state.attributes) return;
+          var curTemp = safeParseFloat(state.attributes.current_temperature);
+          var heating = state.state === 'heat' ? curTemp : null;
```

Comment:
I think we should add the fill for both 'heat
 and 'cool' as separate columns.

However maybe we should check if at least one datapoint exists - otherwise don't add the column.